### DISCUSSION TITLE
Unable to enable apq caching

### DISCUSSION THREAD

**Discussion by esmaeilzadeh** [rebing/graphql-laravel#1073] [Q&A] [answered]
### Versions:
- graphql-laravel Version:8.6.0
- Laravel Version:8.83.27
- PHP Version: 8.0.2

### Description:
I'm using the rebing/graphql-laravel package in my Laravel application to handle GraphQL queries. I have configured Automatic Persisted Queries (APQ) to cache the keys using Redis as the cache driver. However, I'm facing an issue where the keys are not being cached as expected.

In my config/graphql.php file, I have the following APQ configuration:

```
'apq' => [
    'enable' => env('GRAPHQL_APQ_ENABLE', false),
    'cache_driver' => env('GRAPHQL_APQ_CACHE_DRIVER', config('cache.default')),
    'cache_prefix' => config('cache.prefix') . ':graphql.apq',
    'cache_ttl' => 300,
],
```
I have also set the cache driver to Redis in my .env file with the following configuration:

```
GRAPHQL_APQ_ENABLE=true
CACHE_DRIVER=redis
```
According to these settings, the Laravel GraphQL library should cache all queries in the Redis cache database with the prefix laravel_cache:graphql.apq. However, after setting up and running artisan config:cache, I checked both the main database (DB 0) and the cache database (DB 1) in Redis but couldn't find any keys with the laravel_cache:graphql.apq prefix. I confirmed this by running the following command inside the Redis database:

`keys laravel_cache:graphql.apq*
`I have also reviewed the cache configuration in config/cache.php to ensure the Redis cache store is properly defined.

Could you please assist me in resolving this issue? Any help or suggestions would be greatly appreciated.

Thank you in advance.

### Steps To Reproduce:
reproducing this issue is straightforward. normally install the laravel and package and setting the cache to redis and resetting the config by artisan config:cache after that checking redis cache db to see is there any key related to to graphql and there is nothing:
`keys laravel_cache:graphql.apq*`

**Comment by mfn** [comment]
Everything looks fine AFAICS, can you so some debugging if / when certain cache code is called and what it returns?

**Comment by illambo** [comment]
Hi, sorry for the request that may seem trivial, but have you tried to execute some "caching" query (see docs or unit [tests](https://github.com/rebing/graphql-laravel/blob/master/tests/Unit/AutomatedPersistedQueriesTest.php) to look the protocol) ?

**Comment by esmaeilzadeh** [comment, marked as answer]
It seems I had a misunderstanding about the apq. I thought it does not require any configuration from the client side and by default, all queries will become cached after enabling the apq.
